resolver/auth_mutations.resolvers_test.go

```diff
@@ -58,494 +59,369 @@ const (
+func loadLoginTestCases() []loginType {
+	return []loginType{
+		errorFindingUserCase(),
+		errorPasswordValidationCase(),
+		errorActiveStatusCase(),
+		errorFromConfigCase(),
+		errorFromJwtCase(),
+		errorFromGenerateTokenCase(),
+		errorUpdateUserCase(),
+		loginSuccessCase(),
 	}
 }
-
-func handleSpecificTestCase(t *testing.T, tt struct {
-	name     string
-	req      LoginArgs
-	wantResp *fm.LoginResponse
-	wantErr  bool
-	err      error
-}) {
+func TestLogin(
+	t *testing.T,
+) {
+	cases := loadLoginTestCases()
+	// Create a new instance of the resolver
 	resolver1 := resolver.Resolver{}
-	mock, cleanup, _ := testutls.SetupMockDB(t)
-	// Mock database queries based on test case
-	switch tt.name {
-	case ErrorFindingUser:
-		mock.ExpectQuery(regexp.QuoteMeta(`SELECT "users".* FROM "users" WHERE (username=$1) LIMIT 1;`)).
-			WithArgs().
-			WillReturnError(fmt.Errorf(ErrorMsgFindingUser))
-	case ErrorPasswordValidation:
-		rows := sqlmock.NewRows([]string{"id", "password", "active", "role_id"}).
-			AddRow(testutls.MockID, TestPasswordHash, true, 1)
-		mock.ExpectQuery(regexp.QuoteMeta(`SELECT "users".* FROM "users"  WHERE (username=$1) LIMIT 1;`)).
-			WithArgs().
-			WillReturnRows(rows)
-	case ErrorActiveStatus:
-		rows := sqlmock.NewRows([]string{"id", "password", "active", "role_id"}).
-			AddRow(testutls.MockID, OldPasswordHash, false, 1)
-		mock.ExpectQuery(regexp.QuoteMeta(`SELECT "users".* FROM "users"  WHERE (username=$1) LIMIT 1;`)).
-			WithArgs().
-			WillReturnRows(rows)
-	default:
-		rows := sqlmock.NewRows([]string{"id", "password", "active", "role_id"}).
-			AddRow(testutls.MockID, OldPasswordHash, true, 1)
-		mock.ExpectQuery(regexp.QuoteMeta(`SELECT "users".* FROM "users"  WHERE (username=$1) LIMIT 1;`)).
-			WithArgs().
-			WillReturnRows(rows)
-	}
-
-	if tt.name == SuccessCase || tt.name == ErrorUpdateUser {
-		rows := sqlmock.NewRows([]string{"id", "name"}).
-			AddRow(1, "ADMIN")
-		mock.ExpectQuery(regexp.QuoteMeta(`SELECT "roles".* FROM "roles" WHERE ("id" = $1) LIMIT 1`)).
-			WithArgs([]driver.Value{1}...).
-			WillReturnRows(rows)
-	}
-
-	if tt.name == ErrorUpdateUser {
-		mock.ExpectExec(regexp.QuoteMeta(`UPDATE "users" `)).WillReturnError(fmt.Errorf(ErrorMsgfromUpdateUser))
-	} else {
-		result := driver.Result(driver.RowsAffected(1))
-		mock.ExpectExec(regexp.QuoteMeta(`UPDATE "users" `)).WillReturnResult(result)
-	}
-
-	// Execute resolver function
-	c := context.Background()
-	response, err := resolver1.Mutation().Login(c, tt.req.UserName, tt.req.Password)
-
-	// Assert results
-	if tt.wantResp != nil && response != nil {
-		tt.wantResp.RefreshToken = response.RefreshToken
-		tt.wantResp.Token = response.Token
-		assert.Equal(t, tt.wantResp, response)
-	} else {
-		assert.Equal(t, true, strings.Contains(err.Error(), tt.err.Error()))
-		assert.Equal(t, tt.wantErr, err != nil)
+	for _, tt := range cases {
+		t.Run(
+			tt.name,
+			func(t *testing.T) {
+				db, mock, _ := sqlmock.New()
```

Comment:
we should be able to use the setupmockdb function here. We needn't have this here